src/jojoe77777/FormAPI/ModalForm.php

```diff
@@ -23,9 +23,17 @@ public function __construct(?callable $callable) {
         $this->data["button2"] = "";
     }
 
-    public function processData(&$data) : void {
-        if(!is_bool($data)) {
-            throw new FormValidationException("Expected a boolean response, got " . gettype($data));
+    // The error related to the "Esc" key has been fixed
+   
+    public function processData(&$data): void
+    {
+        if ($data === null) {
+            $data = null; 
```

Comment:
This is pointless

Reply:
It prevents you from getting a nasty error in the console. I don't understand what makes sense. Can you give a more detailed answer?

Reply:
If this branch is entered, `$data` is already null, so this assignment is doing nothing useful.

Reply:
What is overlooked in this plugin is that you get an error when you press the escape key. When the data is empty, either yes or no must be selected, so I added something like this to prevent an error if the data is empty.